src/app/[locale]/page.tsx

```diff
@@ -1,15 +1,23 @@
 import {PageWithSidebarContainer} from '@/components/page';
 import datasetFetcher from '@/lib/dataset-fetcher-instance';
-import {useLocale} from 'next-intl';
+import {useLocale, NextIntlClientProvider} from 'next-intl';
 import DatasetList from './dataset-list';
 
 export default async function Home() {
   const initialSearchResult = await datasetFetcher.search();
   const locale = useLocale();
-
+  const messages = (await import(`@/messages/${locale}.json`)).default;
```

Comment:
Curious: are the messages loaded just once or on every invocation of the page? If the latter, is it possible to load the messages at the top level instead?

Reply:
I should be inside the component because you need the `locale` variable. And I only want to use it for this page, not the entire app. So I think this is the correct place. ([see this response](https://github.com/colonial-heritage/dataset-browser/pull/80#discussion_r1106798173))

Reply:
Excellent, thanks!